main/main.cpp

```diff
@@ -1481,7 +1481,6 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 		OS::get_singleton()->_allow_layered = false;
 	}
 
-	OS::get_singleton()->_keep_screen_on = GLOBAL_DEF("display/window/energy_saving/keep_screen_on", true);
```

Comment:
Actually this might change behavior. I don't see the Android implementation querying the setting so it might break now.

I also wonder why it needs to be done manually in each platform instead of just done in `main.cpp` after the DisplaySerer is instantiated?

Reply:
> I don't see the Android implementation querying the setting

Isn't [this the android implementation](https://github.com/godotengine/godot/blob/master/platform/android/java/lib/src/org/godotengine/godot/Godot.java#L292)? I couldn't find any uses for the OS variable at least.

> why it needs to be done manually in each platform

I'm not sure. I could try changing how it's being called, but can't really test it for all platforms.

Reply:
> > I don't see the Android implementation querying the setting
> 
> Isn't [this the android implementation](https://github.com/godotengine/godot/blob/master/platform/android/java/lib/src/org/godotengine/godot/Godot.java?rgh-link-date=2022-08-06T20%3A43%3A46Z#L292)? I also couldn't find any uses for the OS singleton's variable.

Ah indeed.

Ah I was going to suggest making `OS::_keep_screen_on` private but it seems like it's already the case, this was just abusing the fact that `OS` is a friend class to `Main`.